YDBOPS-11054 Install description (#13636)
ydb-platform/ydb: ydb/docs/en/core/reference/ydbops/install.md

{% include [warning.md](_includes/warning.md) %}

## Building from source

1. Clone the `ydbops` repository from GitHub:

    ```bash
    git clone https://github.com/ydb-platform/ydbops.git
    ```

2. There are two ways to build `ydbops`:
    2.1. [Directly with Go](#go)

>     2.2. [Inside a Docker container](#docker)

Author: I think it doesn't look well
<img width="1076" alt="image" src="https://github.com/user-attachments/assets/5ffc816b-f667-4e08-af91-dbc3340047c6" />
Nevertheless, I'll change it

> We are using official docker image for Golang v1.22. It gurantiees successful build. Description Docker container works in the Dockerfile in the repository. The assembly in Docker also performs additional manipulations: linter checks, the version for the ydbops assembly is substituted so that it is registered in the ydbops executable file.

Reviewer: ```suggestion
The second approach requires a prepared [Docker](https://en.wikipedia.org/wiki/Docker_(software)) environment and uses the official Docker image for [Golang](https://en.wikipedia.org/wiki/Go_(programming_language)) v1.22, guaranteeing a successful build. The Docker container operates using the `Dockerfile` from the repository. The build process in Docker also performs additional tasks: running linter checks and substituting the version for the `ydbops` assembly to register it in the executable file.
```

> We are using official docker image for Golang v1.22. It gurantiees successful build. Description Docker container works in the Dockerfile in the repository. The assembly in Docker also performs additional manipulations: linter checks, the version for the ydbops assembly is substituted so that it is registered in the ydbops executable file.

Reviewer: Add a link to the Dockerfile you're talking about.

### Building directly with Go {#go}

#### Prerequsites

> [Install Go](https://go.dev/doc/install). The recommended version is 1.22

Reviewer: ```suggestion
[Install Go](https://go.dev/doc/install). The recommended version is 1.22.
```

#### Compiling

Invoke `go build` in the repository root folder:

```bash
go build
```

> The `ydbops` executable will be available in the repository root folder.
> 
> #### Installing
> 
> You can copy executable file manually or use `make`

Reviewer: ```suggestion
The `ydbops` executable will be available in the repository root folder.

#### Installing

You can copy the executable file manually or use `make`:
```

```bash
make install INSTALL_DIR=install_folder BUILD_DIR=.
```

### Inside a Docker container {#docker}

#### Prerequsites

- make
- [Install docker engine](https://docs.docker.com/engine/install/)

#### Compiling

Invoke this command in the repository root folder:

```bash
make build-in-docker
```

> The `ydbops` executables will be available in the `bin` folder. Binary files are generated for Linux and MacOS (arm64, amd64).
> 
> | Binary name | Platform
> |-|-|
> | ydbops | Linux(amd64) |
> | ydbops_darwin_amd64 | MacOS(amd64) |
> | ydbops_darwin_arm64 | MacOS(arm64) |

Reviewer: ```suggestion
The `ydbops` executables will be available in the `bin` folder. Binary files are generated for Linux and macOS (arm64, amd64).  

| Binary name            | Platform      |
|------------------------|--------------|
| ydbops                | Linux (amd64) |
| ydbops_darwin_amd64   | macOS (amd64) |
| ydbops_darwin_arm64   | macOS (arm64) |
```

> #### Installing
> 
> To install the binary file, execute command `make`.
> 
> Optional parameters:
> 
> - `INSTALL_DIR`: The folder, to which the executable file will be installed. Default value: `~/ydb/bin`.
> 
> - `BUILD_DIR`: The folder that contains the generated executable file. Use this parameter if you created the executable file manually. For example, use `BUILD_DIR=.` if the executable file is in the current working folder.

Reviewer: ```suggestion
#### Installing

To install the binary file, execute the command `make`.

Optional parameters:

- `INSTALL_DIR`: The folder where the executable file will be installed. Default value: `~/ydb/bin`.

- `BUILD_DIR`: The folder that contains the generated executable file. Use this parameter if you created the executable file manually. For example, use `BUILD_DIR=.` if the executable file is in the current working directory.
```

```bash
make install [INSTALL_DIR=<path_to_install_folder>] [BUILD_DIR=<path_to_build_folder>]
```

Sample command to install into `install_folder` from the current folder:

```bash
make install INSTALL_DIR=install_folder BUILD_DIR=.
```

> ## Download the binary from releases page

Reviewer: Let's swap the order between downloading and self-building. People are unlikely to need to compile themselves and might waste time if they miss this download link at the bottom.

> You can download binary releases from [YDBOps Releases](../../downloads/index.md#ydbops).

Reviewer: ```suggestion
You can download binary releases from [{#T}](../../downloads/index.md#ydbops).
```

{% include [warning.md](_includes/warning.md) %}

> ## Download the binary from releases page

Reviewer 2: ```suggestion
## Download the binary from the releases page
```

    git clone https://github.com/ydb-platform/ydbops.git
    ```

3. There are two ways to build `ydbops`:

> - Manual build. Invoke `go build` in the repository root folder:

Reviewer: ```suggestion
- **Manual build.** Invoke `go build` in the repository root folder:
```

    ```bash
    go build
    ```

    The `ydbops` executable will be available in the repository root folder.

> - Automated build. Invoke this command in the repository root folder:

Reviewer: ```suggestion
- **Automated build.** Invoke this command in the repository root folder:
```

    git clone https://github.com/ydb-platform/ydbops.git
    ```

3. There two way to build `ydbops`:

- Manual build. Invoke `go build` in the repository root folder:

    ```bash
    go build
    ```

    The `ydbops` executable will be available in the repository root folder.

> - Automated build. Invoke this command in the the repository root folder:

Reviewer 2: ```suggestion
- Automated build. Invoke this command in the repository root folder:
```

  ```bash
  make build-in-docker
  ```

>   The `ydbops` executables will be available in the `bin` folder. There will be several binary files for Linux, MacOS (arm64, amd64).

Reviewer: It sounds like this will need some extra prerequisites that aren't explained above, probably `make`, `docker`, etc.

I'd also add a small table showing which exactly binaries are generated and which platforms they are for.

Author: ![image](https://github.com/user-attachments/assets/e303ed17-b57c-4e54-87ec-ded892a7cb98)
I've added description for files

4. To install binary file to the specific folder (folder `~/ydb/bin` is defined by default) execute this command

   ```bash
   make install INSTALL_DIR=path_to_folder
   ```

>    You must replace source folder if you have created binary file manually:
> 
>    ```bash
>    make install INSTALL_DIR=путь_к_каталогу BUILD_DIR=.
>    ```

Reviewer 2: ```suggestion
```

## Download the binary from releases page

> You can download binary releases from GitHub - [YDBOps Releases](https://github.com/ydb-platform/ydbops/releases).

Reviewer: We should lead people to https://ydb.tech/docs/en/downloads/#ydbops (and update it regularly) like with the other binaries, not to GitHub.

    git clone https://github.com/ydb-platform/ydbops.git
    ```

3. There are two ways to build `ydbops`:

- Manual build. Invoke `go build` in the repository root folder:

    ```bash
    go build
    ```

    The `ydbops` executable will be available in the repository root folder.

- Automated build. Invoke this command in the repository root folder:

  ```bash
  make build-in-docker
  ```

  The `ydbops` executables will be available in the `bin` folder. Binary files are generated for Linux and MacOS (arm64, amd64).

  | Binary name | Platform
  |-|-|
  | ydbops | Linux(amd64) |
  | ydbops_darwin_amd64 | MacOS(amd64) |
  | ydbops_darwin_arm64 | MacOS(arm64) |

4. To install the binary file, execute command `make`.

   Optional parameters:

  - `INSTALL_DIR`: The folder, to which the executable file will be installed. Default value: `~/ydb/bin`.

>   - `BUILD_DIR`: The folder that contains the generated binary file. Use this parameter if you created the binary file manually. For example, use `BUILD_DIR=.` if the executable file is in the root repository folder.

Reviewer: ```suggestion
  - `BUILD_DIR`: The folder that contains the generated binary file. Use this parameter if you created the binary file manually. For example, use `BUILD_DIR=.` if the executable file is in the current working folder.
```

   ```bash
   make install [INSTALL_DIR=<path_to_install_folder>] [BUILD_DIR=<path_to_build_folder>]
   ```

>    You must replace source folder if you have created binary file manually:

Reviewer: ```suggestion
   You must replace source folder if you have created binary file manually. Example:
```

   ```bash

>    make install INSTALL_DIR=путь_к_каталогу BUILD_DIR=.

Reviewer: ```suggestion
   make install INSTALL_DIR=path_to_install_folder BUILD_DIR=.
```

   ```

## Download the binary from releases page

> You can download binary releases from official site - [YDBOps Releases](https://ydb.tech/docs/en/downloads/#ydbops).

Reviewer: ```suggestion
You can download binary releases from [YDBOps Releases](../../downloads/index.md#ydbops).
```
